conan/tools/gnu/autotoolsdeps.py

```diff
@@ -1,91 +1,63 @@
-                self.sysroot = dep_cpp_info.sysroot
-
-        for d in self._conanfile.dependencies.host.values():
-            merge(d)
+        self._cpp_info = None
+
+    @property
+    def cpp_info(self):
+        if self._cpp_info is None:
+            self._cpp_info = NewCppInfo()
+            for dep in self._conanfile.dependencies.host.values():
+                dep_cppinfo = dep.new_cpp_info.copy()
+                dep_cppinfo.set_relative_base_folder(dep.package_folder)
+                # In case we have components, aggregate them, we do not support isolated
+                # "targets" with autotools
+                dep_cppinfo.aggregate_components()
+                self._cpp_info.merge(dep_cppinfo)
+        return self._cpp_info
 
     def environment(self):
+        flags = AutoToolsDepsFlags(self._conanfile, self.cpp_info)
+
         # cpp_flags
         cpp_flags = []
-        include_paths = ['-I"%s"' % p for p in self.include_paths]
-        cpp_flags.extend(include_paths)
-        cpp_flags.extend(["-D%s" % define for define in self.defines])
-
-        # Libs
-        libs = ["-l%s" % library for library in self.libs]
+        cpp_flags.extend(flags.include_paths)
+        cpp_flags.extend(flags.defines)
 
         # Ldflags
-        # TODO: Discuss, should the helper filter frameworks based on compiler?
-        frameworks = ["-framework %s" % framework for framework in self.frameworks]
-        frameworks_paths = ["-F %s" % framework_path for framework_path in self.framework_paths]
-        ldflags = self.sharedlinkflags
-        ldflags.extend(self.exelinkflags)
-        ldflags.extend(frameworks)
-        ldflags.extend(frameworks_paths)
-        lib_paths = ['-L"%s"' % p for p in self.lib_paths]
-        ldflags.extend(lib_paths)
+        ldflags = flags.sharedlinkflags
+        ldflags.extend(flags.exelinkflags)
+        ldflags.extend(flags.frameworks)
+        ldflags.extend(flags.framework_paths)
+        ldflags.extend(flags.lib_paths)
+        # FIXME: Previously we had an argument "include_rpath_flags" defaulted to False
+        ldflags.extend(flags.rpath_flags)
 
         # cflags
-        cflags = self.cflags
-        cxxflags = self.cxxflags
+        cflags = flags.cflags
+        cxxflags = flags.cxxflags
 
-        if self.sysroot:
-            srf = '--sysroot={}'.format(self.sysroot)
+        srf = flags.sysroot
+        if srf:
             cflags.append(srf)
             cxxflags.append(srf)
             ldflags.append(srf)
 
         env = Environment()
         env.append("CPPFLAGS", cpp_flags)
-        env.append("LIBS", libs)
+        env.append("LIBS", flags.libs)
         env.append("LDFLAGS", ldflags)
         env.append("CXXFLAGS", cxxflags)
         env.append("CFLAGS", cflags)
         return env
 
-    def generate(self):
-        env = self.environment()
-        env.save_sh("conanautotoolsdeps.sh")
-        env.save_bat("conanautotoolsdeps.bat")
+    def generate(self, env=None):
```

Comment:
To discuss: What if we want to modify/add some variable in the environment before generate? what would be the best interface?